src/lib.rs

```diff
@@ -128,6 +132,20 @@ pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     }
     println!("Collected {} stars", stars.len());
 
+    let path = Path::new("output.md");
+    let display = path.display();
+
+    let mut file = match File::create(&path) {
+        Err(why) => panic!("couldn't create {}: {}", 
```

Comment:
`panic!` is generally a last resort, since it causes the program to terminate immediately and generally doesn't print user-friendly output (you can see an example of the unfriendliness in [this Rust playground](https://play.rust-lang.org/?code=%23!%5Ballow(unused)%5D%0A%23!%5Ballow(unreachable_code)%5D%0Afn%20main()%20%7B%0Apanic!()%3B%0Apanic!(%22this%20is%20a%20terrible%20mistake!%22)%3B%0Apanic!(4)%3B%20%2F%2F%20panic%20with%20the%20value%20of%204%20to%20be%20collected%20elsewhere%0Apanic!(%22this%20is%20a%20%7B%7D%20%7Bmessage%7D%22%2C%20%22fancy%22%2C%20message%20%3D%20%22message%22)%3B%0A%7D))

Instead of `match`ing this failure here, we should communicate this failure back into `main()` by returning the `Result`ing `Error` from the current function. The structure in `supernova` follows almost exactly the example CLI tool from [Chapter 12.3](https://doc.rust-lang.org/stable/book/second-edition/ch12-03-improving-error-handling-and-modularity.html#fixing-the-error-handling) of the Rust Book (2nd Ed.), which shows how to refactor a `panic!`-based error approach into printing user-friendly errors, so take a look and see how the example in the Book matches up with the current implementation.

I'd recommend having a look at the [`try!` macro](https://doc.rust-lang.org/std/macro.try.html) and [the `?` operator](https://doc.rust-lang.org/book/second-edition/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator) for different ways to deal with errors in Rust.